controllers/namespace_reconciler.go

```diff
@@ -0,0 +1,90 @@
+package controllers
+
+import (
+	v1 "alexellis/registry-creds/api/v1"
+	"context"
+	"fmt"
+	"strings"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type NamespaceReconciler struct {
```

Comment:
Public types need a comment.

What is the purpose of the `NamespaceReconciler`?